src/webgl/p5.Shader.js

```diff
@@ -16,6 +16,47 @@ var p5 = require('../core/core');
  * will provide the GL context for this new p5.Shader
  * @param {String} vertSrc source code for the vertex shader (as a string)
  * @param {String} fragSrc source code for the fragment shader (as a string)
+ * @example
+ * <div modernizr='webgl'>
+ * <code>
+ * var varying = 'precision highp float; varying vec2 vPos;';
+ * var vs =
+ *   varying +
+ *   'attribute vec3 aPosition;' +
+ *   'void main() { vPos = (gl_Position = vec4(aPosition,1.0)).xy; }';
+ * var fs =
+ *   varying +
+ *   'uniform vec2 p;' +
+ *   'uniform float r;' +
+ *   'const int I = 500;' +
+ *   'void main() {' +
+ *   '  vec2 c = p + vPos * r, z = c;' +
+ *   '  float n = 0.0;' +
+ *   '  for (int i = I; i > 0; i --) {' +
+ *   '    if(z.x*z.x+z.y*z.y > 4.0) {' +
+ *   '      n = float(i)/float(I);' +
+ *   '      break;' +
+ *   '    }' +
+ *   '    z = vec2(z.x*z.x-z.y*z.y, 2.0*z.x*z.y) + c;' +
+ *   '  }' +
+ *   '  gl_FragColor = vec4(0.5-cos(n*17.0)/2.0,0.5-cos(n*13.0)/2.0,0.5-cos(n*23.0)/2.0,1.0);' +
+ *   '}';
+ *
+ * var mandel;
+ * function setup() {
+ *   var renderer = createCanvas(100, 100, WEBGL);
+ *   mandel = new p5.Shader(renderer, vs, fs);
```

Comment:
hmm, other docs say that the return type from createCanvas is an HTMLCanvasElement, not an instance of p5.RendererGL?

Reply:
yup. looks like the docs are wrong ;-)

Reply:
maybe it might be nice to add a `createShader()` method that wraps this functionality?

Reply:
oops :)

there's `loadShader()` but it takes file names! https://p5js.org/reference/#/p5/loadShader, maybe this can be overridden to do what you describe?

documentation for this method needs a good bit of work, but there are some working examples in the manual tests.